src/get-started/flutter-for/ios-devs.md

```diff
@@ -490,200 +859,593 @@ class SignaturePainter extends CustomPainter {
 ```
 
-Then use `CustomButton`,
-just as you'd use any other Flutter widget:
+The following sample generates a list of persons 
+mocked using `mockPersons()`.
+Once the user taps on any person, `pushNamed()` pushes the person's detail page 
+to the `Navigator`.
+
+<nav class="navbar bg-primary">
+ <ul class="navbar-nav navbar-code ml-auto">
+  <li class="nav-item">
+    <a class="btn btn-navbar-code" href="{{site.dartpad}}/?id=5ae0624689958c4775b064d39d108d9e">Test in DartPad</a>
+  </li>
+  <li class="nav-item">
+    <a class="btn btn-navbar-code" href="{{site.repo.this}}/{{sample_path}}/lib/navigation.dart">View on Github</a>
+  </li>
+</ul>
+</nav>
+
+<?code-excerpt "lib/navigation.dart (ListView)"?>
+```dart
+    ListView.builder(
+  itemCount: mockPersons.length,
+  itemBuilder: (context, index) {
+    final person = mockPersons.elementAt(index);
+    final age = '${person.age} years old';
+    return ListTile(
+      title: Text(person.name),
+      subtitle: Text(age),
+      trailing: const Icon(
+        Icons.arrow_forward_ios,
+      ),
+      onTap: () {
+        // when a ListTile that represents a person is
+        // tapped, push the detailsPageRouteName route
+        // to the Navigator and pass the person's instance
+        // to the route
+        Navigator.of(context).pushNamed(
+          detailsPageRouteName,
+          arguments: person,
+        );
+      },
+    );
+  },
+),
+```
 
-<?code-excerpt "lib/custom.dart (UseCustomButton)"?>
+Next you'll define the `DetailsPage` widget responsible for 
+displaying the details of each person. In Flutter, arguments can be 
+dynamically passed into the widget, when navigating to the new route, 
+and extracted using `ModalRoute.of()`:
+
+<nav class="navbar bg-primary">
+ <ul class="navbar-nav navbar-code ml-auto">
+  <li class="nav-item">
+    <a class="btn btn-navbar-code" href="{{site.dartpad}}/?id=5ae0624689958c4775b064d39d108d9e">Test in DartPad</a>
+  </li>
+  <li class="nav-item">
+    <a class="btn btn-navbar-code" href="{{site.repo.this}}/{{sample_path}}/lib/navigation.dart">View on Github</a>
+  </li>
+</ul>
+</nav>
+
+<?code-excerpt "lib/navigation.dart (DetailsPage)"?>
 ```dart
-@override
-Widget build(BuildContext context) {
-  return const Center(
-    child: CustomButton('Hello'),
-  );
+class DetailsPage extends StatelessWidget {
+  const DetailsPage({Key? key}) : super(key: key);
+
+  @override
+  Widget build(BuildContext context) {
+    // read the person instance from the arguments
+    final Person person = ModalRoute.of(
+      context,
+    )?.settings.arguments as Person;
+    // extract the age
+    final age = '${person.age} years old';
+    return Scaffold(
+      // display name and age
+      body: Column(children: [Text(person.name), Text(age)]),
+    );
+  }
 }
 ```
 
-## Navigation
+For more advanced navigation and routing requirements, 
+you should use a routing package such as [go_router][]. 
+You can find more details at Navigation in Flutter.  
```

Comment:
Add a link.

Reply:
This was added in a previous commit: https://github.com/flutter/website/pull/7736/commits/05e1e46a2aac387685280d74c4dd9ffc61660595